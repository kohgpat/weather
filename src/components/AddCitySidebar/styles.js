import styled from "styled-components";
import ButtonComponent from "../Button";

export const Sidebar = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  width: 300px;
  background-color: #fff;
  display: none;
  z-index: 1;
  border-left: var(--add-city-sidebar-border);

  ${props => props.isVisible && `
    display: flex;
    flex-direction: column;
  `}
`;

export const SidebarHeader = styled.header`
  display: flex;
  padding: 20px;
`;

export const SidebarCloseButton = styled(ButtonComponent).attrs({ type: "button" })`
  width: auto;
  margin-left: auto;
`;
