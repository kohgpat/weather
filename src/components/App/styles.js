import styled from "styled-components";

export const App = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
`;
