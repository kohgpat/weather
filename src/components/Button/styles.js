import styled from "styled-components";

export const Button = styled.button`
  font-size: 14px;
  font-weight: bold;
  font-family: "Roboto Condensed", "Helvetica Neue", "Helvetica", Arial, sans-serif;
  padding: 8px 16px;
  height: 40px;
  border: 0;
  background-color: #eee;
  width: 100%;
`;
