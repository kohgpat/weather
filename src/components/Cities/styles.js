import styled from "styled-components";

export const Cities = styled.div`
  margin-top: calc(50px + var(--topbar-height));
  padding: 20px;

  @supports (display: grid) {
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    grid-gap: 30px;
    grid-auto-rows: minmax(300px, auto);

    @media (min-width: 768px) {
      grid-template-columns: repeat(2, 1fr);
    }

    @media (min-width: 1028px) {
      grid-template-columns: repeat(3, 1fr);
    }
  }

  @supports not (display: grid) {
    display: flex;
    flex-direction: row;
    flex-shrink: 0;
    align-items: center;
    justify-content: center;
    width: 100%;
    flex-wrap: wrap;
  }
`;

export const NoCitiesMessage = styled.div`
  margin-top: calc(50px + var(--topbar-height));
  padding: 20px;
`;
