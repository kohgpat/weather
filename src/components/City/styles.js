import styled from "styled-components";

export const City = styled.div`
  height: 280px;
  width: 280px;
  background-color: #fff;
  border-radius: 4px;
  box-shadow: 0 0 20px -2px rgba(0,0,0,0.1);
  display: flex;
  flex-direction: column;
  padding: 10px;

  @media (min-width: 768px) {
    height: 300px;
    width: 300px;
  }

  @supports not (display: grid) {
    margin-top: 30px;

    @media (orientation: landscape) {
      &:not(:last-child) {
        margin-right: 30px;
      }
    }

    @media (min-width: 768px) {
      &:not(:last-child) {
        margin-right: 30px;
      }
    }
  }
`;

export const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Name = styled.h1`
  margin: 10px;
  width: 80%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
`;

export const RemoveButton = styled.button.attrs({ type: "button" })`
  padding: 0;
  font-weight: bold;
  font-size: 32px;
  background-color: transparent;
  border: 0;
  margin: 10px 10px 10px auto;
  visibility: visibility;
  cursor: pointer;

  @media (min-width: 768px) {
    visibility: hidden;

    ${City}:hover & {
      visibility: visible;
    }
  }
`;

export const WeatherIcon = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: auto 0;
`;

export const Stats = styled.div`
  margin: 10px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Stat = styled.h4`
  text-transform: capitalize;
  margin: 0;

  ${props =>
    props.major &&
    `
    font-size: 3rem;
  `};
`;
