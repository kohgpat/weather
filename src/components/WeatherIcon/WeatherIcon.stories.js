/* global module */
import React from "react";
import { storiesOf } from "@storybook/react";
import WeatherIcon from "./index";

storiesOf("WeatherIcon", module).add("default", () => (
  <WeatherIcon weather={{ icon: "rain.png" }} />
));
