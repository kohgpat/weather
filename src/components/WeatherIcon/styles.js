import styled from "styled-components";

const ENDPOINT = "https://openweathermap.org/img/w/";

export const Icon = styled.div`
  width: 50px;
  height: 50px;

  ${props =>
    props.icon &&
    `
    background-image: url(${ENDPOINT}${props.icon}.png);
  `};
`;
