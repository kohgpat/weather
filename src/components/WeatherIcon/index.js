import React from "react";
import "weather-icons/css/weather-icons.min.css";
import * as styles from "./styles";

const isDayTime = () => {
  const hours = new Date().getHours();
  const isDayTime = hours > 6 && hours < 20;

  return isDayTime;
};

const WeatherIcon = ({ weather }) => {
  const isDay = isDayTime();

  let wiClass = "";

  if (weather.description === "mist") {
    wiClass += isDay ? " wi-day-fog" : " wi-night-fog";
  }

  if (weather.description === "broken clouds" ||
      weather.description === "few clouds") {
    wiClass += isDay ? " wi-day-cloudy" : " wi-night-cloudy";
  }

  if (weather.description === "clear sky") {
    wiClass += isDay ? " wi-day-sunny" : " wi-night-clear";
  }

  if (weather.description === "overcast clouds") {
    wiClass += " wi-day-sunny-overcast";
  }

  if (weather.description === "light rain" ||
      weather.description === "shower rain") {
    wiClass += isDay ? " wi-day-rain" : " wi-night-rain";
  }

  if (weather.description === "smoke") {
    wiClass += " wi-smoke";
  }

  return (
    <styles.Icon className={`wi ${wiClass}`} />
  )
};

export default WeatherIcon
