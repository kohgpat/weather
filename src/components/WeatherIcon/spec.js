import React from "react";
import WeatherIcon from "./index";
import * as styles from "./styles";
import { shallow, render } from "enzyme";

const weather = {
  icon: "10d"
};

it("renders", () => {
  const iconWithoutWeather = shallow(<WeatherIcon />);
  expect(iconWithoutWeather).toMatchSnapshot();

  const iconWithWeather = shallow(<WeatherIcon weather={weather} />);
  expect(iconWithWeather).toMatchSnapshot();
});

it("renders null without weather", () => {
  const icon = shallow(<WeatherIcon />);
  expect(icon.type()).toEqual(null);
});
