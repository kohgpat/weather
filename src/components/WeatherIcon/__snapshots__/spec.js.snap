// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `""`;

exports[`renders 2`] = `
<styled.div
  icon="10d"
/>
`;
